### PR TITLE
ci: only publish node packages on changes

### PR DESCRIPTION
**Pull request by dschaller**
### Description
Only attempt to publish npm packages if they don't exist in the registry.

```bash
lerna info Executing command in 10 packages: "yarn run publishAlpha"
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/tools' in 1.3s:
$ ../../../tools/publish-frontend.sh tools
Found existing version of tools@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/core' in 0.9s:
$ ../../../tools/publish-frontend.sh core
Found existing version of core@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/data-layout' in 1.2s:
$ ../../../tools/publish-frontend.sh data-layout
Found existing version of data-layout@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/wizard' in 1.1s:
$ ../../../tools/publish-frontend.sh wizard
Found existing version of wizard@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/kinesis' in 1.5s:
$ ../../../tools/publish-frontend.sh kinesis
Found existing version of kinesis@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/k8s' in 1.5s:
$ ../../../tools/publish-frontend.sh k8s
Found existing version of k8s@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/experimentation' in 1.5s:
$ ../../../tools/publish-frontend.sh experimentation
Found existing version of experimentation@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/sourcecontrol' in 1.6s:
$ ../../../tools/publish-frontend.sh sourcecontrol
Found existing version of sourcecontrol@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/ec2' in 1.6s:
$ ../../../tools/publish-frontend.sh ec2
Found existing version of ec2@0.0.0-20200723151653-124f38193eb4
lerna info run Ran npm script 'publishAlpha' in '@clutch-sh/envoy' in 1.6s:
$ ../../../tools/publish-frontend.sh envoy
Found existing version of envoy@0.0.0-20200723151653-124f38193eb4
lerna success run Ran npm script 'publishAlpha' in 10 packages in 5.2s:
```

### CODE REVIEW

File: tools/publish-frontend.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+VERSION="0.0.0-$(git log -1 --format=%cd --date=format:'%Y%m%d%H%M%S')-$(git rev-parse --short=12 HEAD)"
+PACKAGE=$1
+
+if yarn info "@clutch-sh/${PACKAGE}" | grep -q "$VERSION"; then
+  echo "Found existing version of ${PACKAGE}@${VERSION}"
+  exit 0
+fi
+
+(
```
Comment (danielhochman) [inline tools/publish-frontend.sh:11]: why is this in a subshell?

File: tools/publish-frontend.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  cd "$PWD" && yarn publish --new-version="$VERSION" --access public --no-git-tag-version
```
Comment (danielhochman) [inline tools/publish-frontend.sh:12]: cd $PWD is redundant i think

**Comment by danielhochman** [comment]
@dschaller ping on this or should we close?

**Comment by dschaller** [comment]
I'll resurrect this

File: tools/publish-frontend.sh (hunk 1 above)
Comment (dschaller) [reply tools/publish-frontend.sh]: Removed

**Comment by dschaller** [comment]
cla check

File: tools/publish-frontend.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+VERSION="0.0.0-beta.$(git log -1 --format=%cd --date=format:'%Y%m%d%H%M%S')"
+PACKAGE=$1
+
+if yarn info "@clutch-sh/${PACKAGE}" | grep -q "$VERSION"; then
+  echo "Found existing version of ${PACKAGE}@${VERSION}"
+  exit 0
+fi
+
+yarn pack --new-version="${VERSION}" --access public --no-git-tag-version
```
Comment (danielhochman) [inline tools/publish-frontend.sh:11]: yarn pack? should this be yarn publish?

Comment (dschaller) [reply tools/publish-frontend.sh]: Ah yeah. Sorry was using pack for testing :)

File: tools/publish-frontend.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+VERSION="0.0.0-beta.$(git log -1 --format=%cd --date=format:'%Y%m%d%H%M%S')"
+PACKAGE=$1
+
+if yarn info "@clutch-sh/${PACKAGE}" | grep -q "$VERSION"; then
+  echo "Found existing version of ${PACKAGE}@${VERSION}"
+  exit 0
+fi
+
+yarn publish --new-version="${VERSION}" --access public --no-git-tag-version
```
Comment (mcutalo88) [inline tools/publish-frontend.sh:11]: How does this know what package is going to publish? Just based on CWD?

Comment (danielhochman) [reply tools/publish-frontend.sh:11]: yep, should be

**Pull request merged**